controllers/register.go

```diff
@@ -34,24 +41,24 @@ func (r *RegisterController) register(c *gin.Context) {
 
 	if err := c.ShouldBind(form); err != nil {
 		for _, fieldErr := range err.(validator.ValidationErrors) {
-			c.HTML(http.StatusBadRequest, "register", gin.H{
-				"title": "Register",
-				"error": errorHandler.ValidationErrorToText(fieldErr),
-				"email": form.Email,
-			})
+			renderRegisterWithError(c, http.StatusBadRequest, errorHandler.ValidationErrorToText(fieldErr), form)
 			return
 		}
 	}
 
 	if err := models.SaveUser(form.Email, form.Password); err != nil {
-		c.HTML(http.StatusBadRequest, "register", gin.H{
-			"title": "Register",
-			"error": errorHandler.DatabaseErrorToText(err),
-			"email": form.Email,
-		})
+		renderRegisterWithError(c, http.StatusBadRequest, errorHandler.DatabaseErrorToText(err), form)
 		return
 	}
 
-	session.AddFlash(c, "Register successfully")
-	c.Redirect(http.StatusFound, "/")
+	session.AddFlash(c, registerSuccessfully)
+	c.Redirect(http.StatusFound, "/login")
+}
+
+func renderRegisterWithError(c *gin.Context, status int, errMsg string, form *RegisterForm) {
```

Comment:
That could be moved to a helper file s it will be used on more than one controller?

Reply:
I implemented this way because I don't know how to add another values (depend on page) to `gin.H` (2 required values for any error html render are `title` and `errors`)  for example the register (another value for register page is `email`)
```go
func RenderErrorHTML(c *gin.Context, status int, view string, title string, err error, anotherValues []interface{}) {
  data = gin.H {
    "titles": title,
    "errors": err,
    ... iterate from anotherValues
  }
  // completed data should be like this.
  // data = gin.H {
  //  "titles": title,
  //  "errors": err,
  //  "email": ...,
  // }

   ginview.HTML(c, status, view, data)
}
```

We can create helper function for current implementation and pass the whole `gin.H` into it something like this
```go
package helpers

func RenderHTML(c *gin.Context, status int, view string, data gin.H) {
  ginview.HTML(c, status, view, data)
}
```

but I think this is redundant as we can just use `ginview.HTML(...)` directly instead wrap this in another function.

What do you think?

Reply:
The first example you have shown is what I had in mind: 

```go
func renderWithError(c *gin.Context, status int, view string, title string, err error, rest ...[]string{}) {
    default_params = gin.H{
		"title":  title,
		"errors": err.Error(),
	}
    view_params = append(default_params, rest...)
	ginview.HTML(c, status, view, view_params)
}
```

Untested though 😅 but that's the idea.

Reply:
`append` won't work with `gin.H` because it is not slice.

I implemented this way instead d6739d5fc316f5743612313078207bb0c9d8e2c4.

What do you think?

Reply:
Applied above solution to home and login in 458cd93353f3366cfc2bc03932583c408d577b8e

Reply:
It's an improvement to keep things DRY. I am not usually a fan of using `data` as a variable name. Phoenix (Elixir) calls this `assigns` and Rails calls this `locals` FYI. `data` works but it's not descriptive at all 🧐